test/e2e/nodemaintenance_test.go

```diff
@@ -151,27 +173,157 @@ func nodeMaintenanceTest(t *testing.T, f *framework.Framework, ctx *framework.Te
 		},
 	}
 
-	// Create node maintenance CR
+// Create node maintenance CR
 	err = f.Client.Create(goctx.TODO(), nodeMaintenance, &framework.CleanupOptions{TestContext: ctx, Timeout: cleanupTimeout, RetryInterval: cleanupRetryInterval})
+	if err != nil {
+		t.Fatalf("Can't create CRD: %v", err)
+	}
+
+	// Get Running phase first
+	if err := wait.PollImmediate(1*time.Second, 20*time.Second, func() (bool, error) {
+		nm := &operator.NodeMaintenance{}
+		err := f.Client.Get(goctx.TODO(), types.NamespacedName{Name: "nodemaintenance-xyz"}, nm)
+		if err != nil {
+			t.Logf("Failed to get %q nodeMaintenance: %v", nm.Name, err)
+			return false, nil
+		}
+
+		//t.Logf("%s: operator running state: %s", time.Now().Format("2006-01-02 15:04:05.000000"), nm.Status.Phase)
+		if nm.Status.Phase != operator.MaintenanceRunning {
+			t.Logf("%s: Running", time.Now().Format("2006-01-02 15:04:05.000000"))
+			return false, nil
+		}
+
+		if nm.Status.LastError != nmo.QuorumViolationErrorMsg {
+			t.Logf("%s: Running, with wrong error: %s", time.Now().Format("2006-01-02 15:04:05.000000"), nm.Status.LastError)
+			return false, nil
+		}
+		return true, nil
+	}); err != nil {
+		showDeploymentStatus(t, f, fmt.Errorf("%s: Failed to verify running phase: %v", time.Now().Format("2006-01-02 15:04:05.000000"), err))
+	}
+
+	t.Logf("correct error has been observed")
```

Comment:
So the e2e tests never actually succeed in putting node into maintenance mode anymore?